### PR TITLE
CD-401

### PR DESCRIPTION
**Pull request by PivaFiras**
CD-401
- Add Cradlepoint NetCloud Manager - Network Device Status script
- Add script description JSON

### CODE REVIEW

**Review by mohamedkallel — APPROVED** [review]

File: examples/http/cradlepoint_manager_network_devices_status.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
+    "logo": "",
```
Comment (GeorgiTsatsev) [inline examples/http/cradlepoint_manager_network_devices_status.json:2]: missing logo

File: examples/http/cradlepoint_manager_network_devices_status.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "version": "1.0.0",
+    "name": "Cradlepoint NetCloud Manager - Network Device Status",
+    "description": "Retrieves the status of network devices for monitoring.",
+    "category": "Software Integration",
+    "requirements": {
```
Comment (GeorgiTsatsev) [inline examples/http/cradlepoint_manager_network_devices_status.json:7]: missing sandbox_version requirement

File: examples/http/cradlepoint_manager_network_devices_status.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            "name": "X_ECM_API_ID",
+            "label": "Enterprise Content Management API ID"
+        },
+        {
+            "value_type": "SECRET_TEXT",
+            "name": "X_ECM_API_KEY",
+            "label": "Enterprise Content Management API Key"
+        }
+    ]
+}
```
Comment (GeorgiTsatsev) [inline examples/http/cradlepoint_manager_network_devices_status.json:50]: missing tested_on

File: examples/http/cradlepoint_manager_network_devices_status.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    "expected_variables": {
+        "table": "[3]"
+    },
+    "protocols": [
+        "HTTP"
+    ],
+    "parameters": [
+        {
+            "value_type": "STRING",
+            "name": "deviceIds",
```
Comment (GeorgiTsatsev) [inline examples/http/cradlepoint_manager_network_devices_status.json:26]: Why is this pluar but a string?
Should it not be 'deviceId' with a list type?

File: examples/http/cradlepoint_manager_network_devices_status.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+var headers = {
+    'X-CP-API-ID': X_CP_API_ID,
+    'X-CP-API-KEY': X_CP_API_KEY,
+    'X-ECM-API-ID': X_ECM_API_ID,
+    'X-ECM-API-KEY': X_ECM_API_KEY
+}
+
+var url =  "/api/v2/net_devices";
+if (deviceIdsFilter.length !== 1 && deviceIdsFilter[0].toLowerCase() !== 'all') {
+    url += "/?id__in=" + deviceIdsFilter.join(',');
```
Comment (GeorgiTsatsev) [inline examples/http/cradlepoint_manager_network_devices_status.js:33]: is the double __ really correct?

File: examples/http/cradlepoint_manager_network_devices_status.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+var headers = {
+    'X-CP-API-ID': X_CP_API_ID,
+    'X-CP-API-KEY': X_CP_API_KEY,
+    'X-ECM-API-ID': X_ECM_API_ID,
+    'X-ECM-API-KEY': X_ECM_API_KEY
+}
+
+var url =  "/api/v2/net_devices";
+if (deviceIdsFilter.length !== 1 && deviceIdsFilter[0].toLowerCase() !== 'all') {
```
Comment (GeorgiTsatsev) [inline examples/http/cradlepoint_manager_network_devices_status.js:32]: Is all an option? Its not stated in the meta data at all

File: examples/http/cradlepoint_manager_network_devices_status.js [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+ * @documentation This procedure is used for retrieving device * variables data
+ */
+function get_status() {
+    getNetworkDevicesStatus()
+        .then(function (bodyResponse) {
+            var networkDevicesStatus = bodyResponse.data;
+            if (!networkDevicesStatus.length) {
+                console.info('There are no network devices status related to this filter.');
+            }
+            if (!Array.isArray(networkDevicesStatus) && typeof networkDevicesStatus === 'object') {
```
Comment (GeorgiTsatsev) [inline examples/http/cradlepoint_manager_network_devices_status.js:101]: wouldn't that be an else?

**Review by GeorgiTsatsev — APPROVED** [review]

**Review by GeorgiTsatsev — APPROVED** [review]

**Pull request merged**